volatility3/framework/plugins/windows/mbrscan.py

```diff
@@ -0,0 +1,109 @@
+                                                                'PARTITION_ENTRY': mbr.PARTITION_ENTRY
+                                                               })
+
+        partition_table_object = symbol_table + constants.BANG + "PARTITION_TABLE"
+        
+        # Define Signature and Data Length
+        mbr_signature = b"\x55\xAA"
+        mbr_length = 0x200
+        bootcode_length = 0x1B8
+
+        # Scan the Layer for Raw Master Boot Record (MBR) and parse the fields
+        for offset, _value in layer.scan(context = self.context, scanner = scanners.MultiStringScanner(patterns = [mbr_signature])):
+            try:
+                mbr_start_offset = offset - (mbr_length - len(mbr_signature))
+                partition_table = self.context.object(partition_table_object, offset = mbr_start_offset, layer_name = layer.name)
+
+                # Extract only BootCode
+                full_mbr = layer.read(mbr_start_offset, mbr_length, pad = True)
+                bootcode = full_mbr[:bootcode_length]
+                
+                if bootcode:
+                    all_zeros = bootcode.count(b"\x00") == len(bootcode)
+
+                if not all_zeros:
+                    partition_entries = [ 
+                                            partition_table.FirstEntry,
+                                            partition_table.SecondEntry,
+                                            partition_table.ThirdEntry,
+                                            partition_table.FourthEntry
+                                        ]
+                    partition_info = "\n"
+
+                    for index, partition_entry_object in enumerate(partition_entries):
+                        partition_entry_object.set_index(index)
+                        partition_info += str(partition_entry_object)
+                    
+                    yield 0, (
+                            format_hints.Hex(offset),
+                            partition_table.get_disk_signature(),
+                            self.get_hash(bootcode),
+                            self.get_hash(full_mbr),
+                            partition_info,
+                            interfaces.renderers.Disassembly(bootcode, 0, architecture),
+                            format_hints.HexBytes(bootcode)
+                    )
+                else:
+                    vollog.log(constants.LOGLEVEL_VV, f"Not a valid MBR: Data all zeroed out : {format_hints.Hex(offset)}")
+            
+            except exceptions.PagedInvalidAddressException:
+                pass
+
+    def run(self):
+        return renderers.TreeGrid([
+            ("Potential MBR at Physical Offset", format_hints.Hex),
+            ("Disk Signature", str),
+            ("Bootcode md5", str),
+            ("Bootcode (FULL) md5", str),
+            ("Partition Entries Info", str),
+            ("Disasm", interfaces.renderers.Disassembly),
+            ("Hexdump", format_hints.HexBytes)
```

Comment:
If the disassembly code isn't present, it will return the hexbytes, and if it does, the bytes should be output as part of the disassembly, so I'm not certain having this column makes sense.

Reply:
The sources of both data are the same, but I think we need both the **Disassembly** format to track the actual MBR behavior and the **Hexdump** format columns to check data such as MBR Error Messages (Invalid partition tableError loading operating system Missing operating system). [Volatility2 ](https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/mbrparser.py#L453) seems to be dealing with this optionally. What do you think about this? If I misunderstand this feedback, please tell me in more detail.
```python
if not self._config.HEX:
                disasm = self.get_disasm_text(boot_code, start + dis)
                if disasm == "" or self.code_data == None:
                    continue
                boot_code_output = "Disassembly of Bootable Code:\n{0}\n\n".format(disasm)
            else:
                hexstuff = "\n" + "\n".join(["{0:010x}  {1:<48}  {2}".format(o, h, ''.join(c)) for o, h, c in self.Hexdump(boot_code, start)])
                boot_code_output = "Bootable code: \n{0} \n\n".format(hexstuff)
```

Reply:
I had thought that the disassembly would contain the hexdump, but perhaps it doesn't.  If you think it's important that the hexdump is always in there, then I'm ok with you leaving it, but just to let you know that users without a disassembler will see the hexdump twice.

Reply:
I beg to differ. Check out the example display. It's highly useful to also see the hexdump, since it typically carries strings, that help to identify the source of it. E.g. a Microsoft bootloader looks different to a grub2 bootloader, but those would be hard to differentiate with the disassembly only (that not even tries to separate data from code blocks, etc..). 

I would even like to ask for an (hex) offset column for the hexdump, which would allow to logically syncronize both views.

Hiding this behind a `--full` option is fine of course.